cql3/statements/batch_statement.cc

```diff
@@ -75,12 +75,12 @@ batch_statement::batch_statement(type type_,
 
 bool batch_statement::depends_on_keyspace(const sstring& ks_name) const
 {
-    return false;
+    return boost::algorithm::any_of(_statements, [&ks_name] (auto&& s) { return s.statement->depends_on_keyspace(ks_name); });
 }
 
 bool batch_statement::depends_on_column_family(const sstring& cf_name) const
 {
-    return false;
+    return boost::algorithm::any_of(_statements, [&cf_name] (auto&& s) { return s.statement->depends_on_column_family(cf_name); });
```

Comment:
This doesn't work well with the code that calls it in query_processor::migration_subscriber::should_invalidate. If ks1.cf is changed, statements that depend on ks2.cf will be invalidated.

Reply:
I think it doesn't make sense to have a function `depends_on_column_family(cf_name)` because cf_name is meaningless without knowing in which keyspace... I think the function needs to be changed to take both the ks and the cf, not just the cf.

Reply:
We need: depends_on(string_view ks, optional<string_view> table)

Reply:
yes, it will invalidate if the tested (ks,cf) pair is in the cartesian product (ks deps)x(cf deps) it is all wrong and I should change it. I think I'll follow @nyh sugestion and ad the ks to the depends on column family.